pkg/hvs/tasks/create_default_template.go

```diff
@@ -0,0 +1,460 @@
+	var templates []hvs.FlavorTemplate
+
+	ftStore, err := t.flavorTemplateStore()
+	if err != nil {
+		return errors.Wrap(err, "Failed to initialize flavor template store instance")
+	}
+
+	if len(t.deleted) != 0 {
+		// Recover deleted default template.
+		err := ftStore.Recover(t.deleted)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to recover default flavor template(s) %s", t.deleted)
+		}
+		t.deleted = []string{}
+		return nil
+	}
+
+	templates, err = getTemplates()
+	if err != nil {
+		return err
+	}
+
+	for _, ft := range templates {
+		_, err := ftStore.Create(&ft)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create default flavor template with ID \""+ft.ID.String()+"\"")
+		}
+	}
+
+	return nil
+}
+
+func (t *CreateDefaultTemplate) Validate() error {
+	ftStore, err := t.flavorTemplateStore()
+	if err != nil {
+		return errors.Wrap(err, "Failed to initialize flavor template store instance")
+	}
+	var ftList []hvs.FlavorTemplate
+	defaultFlavorTemplateMap := map[string]bool{}
+	t.deleted = []string{}
+
+	for _, templateName := range defaultFlavorTemplateNames {
+		defaultFlavorTemplateMap[templateName] = false
+	}
+
+	ftList, err = ftStore.Search(false)
+	if err != nil {
+		return errors.Wrap(err, "Failed to validate "+t.commandName)
+	}
+	if len(ftList) == 0 {
+		return errors.New("No active templates found in db")
+	}
+
+	for _, template := range ftList {
+		defaultFlavorTemplateMap[template.Label] = true
+	}
+
+	for _, templateName := range defaultFlavorTemplateNames {
+		if !defaultFlavorTemplateMap[templateName] {
+			t.deleted = append(t.deleted, templateName)
+		}
+	}
+
+	if len(t.deleted) != 0 {
+		return errors.New(t.commandName + ": Failed to create default flavor template(s) \"" + strings.Join(t.deleted, " "))
+	}
+	return nil
+}
+
+func (t *CreateDefaultTemplate) PrintHelp(w io.Writer) {
+	setup.PrintEnvHelp(w, DbEnvHelpPrompt, "", DbEnvHelp)
+	fmt.Fprintln(w, "")
+}
+
+func (t *CreateDefaultTemplate) SetName(n, e string) {
+	t.commandName = n
+}
+
+func (t *CreateDefaultTemplate) flavorTemplateStore() (*postgres.FlavorTemplateStore, error) {
+	var dataStore *postgres.DataStore
+	var err error
+	if t.TemplateStore == nil {
+		dataStore, err = postgres.NewDataStore(postgres.NewDatabaseConfig(constants.DBTypePostgres, &t.DBConf))
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to connect database")
+		}
+		t.TemplateStore = postgres.NewFlavorTemplateStore(dataStore)
+	}
+	if t.TemplateStore.Store == nil {
+		return nil, errors.New("Failed to create FlavorTemplateStore")
+	}
+	return t.TemplateStore, nil
+}
+
+func getTemplates() ([]hvs.FlavorTemplate, error) {
+	var ret []hvs.FlavorTemplate
+
+	for _, ftStr := range defaultFlavorTemplatesRaw {
+		var ft hvs.FlavorTemplate
+		err := json.Unmarshal([]byte(ftStr), &ft)
+		if err != nil {
+			return nil, err
+		}
+		ret = append(ret, ft)
+	}
+	return ret, nil
+}
+
+var defaultFlavorTemplatesRaw = []string{
```

Comment:
Can we create separate files for the templates in our repo so that they can be maintained them as code (we expect there will be changes needed)?  @ddhawal can you suggest a location for them?

Reply:
If these are used only in tasks we can go with - 
in case of multiple templates - tasks/resources/templates
if there is only on the template then - tasks/resources/

Reply:
Added the templates as separate files in  the location : tasks/resources/templates.